docs/DATA.md

```diff
@@ -0,0 +1,88 @@
+
+Namespace conflicts can arise if two different applications choose to write to the same namespace. `TODO we need to figure out a way to resolve this`.
+
+## Schemas and Sources
+Given the ability for many different applications to write data across many different types of datastores, things will quickly grow to complex for an ecosystem of interoperable applications to thrive.
+
+To support data interoperability, Habitat provides an API for specifying common APIs that communities can implement. This system is split into three components: 
+* Schemas
+* Interfaces
+* Implementations
+
+The purpose of this is to make it easy to tell what APIs a community actually publishes. In a sentence, you might say `"community X implements interface Y which returns schema Z"`. 
+
+`Schemas` are structured data types that are registered with the `data_system_name`. They are content-addressed, so changes to the structure will result in a reference to a fundamentally different schema. They are written in the format of [JSON Schemas](https://json-schema.org/).
+
+`Interfaces` pair a schema with a description of an API being provided. There can be many `Interfaces` with the same `Schema`, as long as their description text differs. The reason for this is that many APIs can serve the same data type while having completely different semantics. For example, if there was a `photo` schema, two very different interfaces for it might be `profile_picture` and `receipt_picture`. The underlying type is the same, but what the API is specifying is completely different.
+
+Each community can maintain a list of `Implementations` for `Interfaces` that it cares about. A `Implementation` is just a path to a specific piece of data following the format `/<community_id>/<datastore>/<namespace>/<content_identifer>` (just the back half of the URL to access any given data in the first place). Note that a community can provide multiple `Implementations` for a single `Interface`. This is to allow different datastores, and different content identifiers within the same store to be able to implement an `Interface`. It is up to applications to choose which `Implementation` to use if multiple are presented. Since not all datastores present their data in a format that converts cleanly to JSON, each datastore must provide a facility to "JSONify" data retrieved via a content identifier. More advanced custom transformations can be defined in the future as well.
```

Comment:
So apps implement the backing store, but communities serve them. so they keep a list of `implementations`. I need to think about this a little more.

Reply:
Yeah. The idea is to have different datastores be able to serve data conforming to the same schema

Reply:
how does this work with SQL type datastores or even like Mongo DB? then the data isn't really stored in a path

is it possible to come up with a design where any backing data store could be plug and play?